### PR TITLE
HSEARCH-2998 + HSEARCH-2992 Get rid of settings-example.xml

### PR DESCRIPTION
**Pull request by yrodiere**
* https://hibernate.atlassian.net//browse/HSEARCH-2998
* https://hibernate.atlassian.net//browse/HSEARCH-2992

Tested on CI: http://ci.hibernate.org/view/Search/job/hibernate-search-master-mavencentral/13/

### CODE REVIEW

**Comment by yrodiere** [comment]
@Sanne at long last... It works: http://ci.hibernate.org/view/Search/job/hibernate-search-master-mavencentral/13/
Note the PR build failed because the settings-example.xml doesn't exist anymore. I will change the job's configuration once this PR is merged.

File: integrationtest/osgi/karaf-it/src/test/java/org/hibernate/search/test/integration/osgi/HibernateSearchWithKarafIT.java [hunk 1]
```diff
@@ -121,6 +123,9 @@
 				.type( "xml" )
 				.versionAsInProject();
 
+		Properties mavenProperties = new Properties();
+		mavenProperties.load( getClass().getResourceAsStream( "/maven.properties" ) );
```
Comment (Sanne) [inline integrationtest/osgi/karaf-it/src/test/java/org/hibernate/search/test/integration/osgi/HibernateSearchWithKarafIT.java]: Shouldn't this stream be closed?

**Comment by Sanne** [comment]
Very nice, and the productization team will be pleased as well, as they are able to override our hard-coded stuff at build time now :+1:

File: integrationtest/osgi/karaf-it/src/test/java/org/hibernate/search/test/integration/osgi/HibernateSearchWithKarafIT.java (hunk 1 above)
Comment (Sanne) [reply integrationtest/osgi/karaf-it/src/test/java/org/hibernate/search/test/integration/osgi/HibernateSearchWithKarafIT.java]: don't bother, I've changed it locally... testing, if it's green I'll merge.

Comment (yrodiere) [reply integrationtest/osgi/karaf-it/src/test/java/org/hibernate/search/test/integration/osgi/HibernateSearchWithKarafIT.java]: Ha, right, that was dirty work... Thanks :)

**Comment by Sanne** [comment]
merged

**Pull request closed without merging**